### PR TITLE
Tollerate %xx in filenames

### PR DESCRIPTION
**Pull request by yarikoptic**
To address #4950 , but also introducing `%b4` into an obscure filename used by tests. Sits on top of #4949 .  The actual fix in the last commit in the series was trivial and to the one of the most ancient pieces of code in the archive (current commit is f080379e8ae6faf61c0927662940399be9aa5543 ).  Let's now just see if tests pass to signal that there were no some hidden assumption somewhere ;-)

### CODE REVIEW

**Review by kyleam — APPROVED** [review]
> Subject: [PATCH 4/5] ENH: add %b5 into parts of an obscure filename
>
> So we could verify that filenames are not unquoted where they should not be

Okay, so this shows up as failures in `test_archives.test_basic_scenario` and `test_archives.test_annex_get_from_subdir`.  And ...

> Subject: [PATCH 5/5] BF: do not urlunquote filename of a file from an archive

... both these failures are addressed by this commit.

**Comment by yarikoptic** [comment]
<details>
<summary>travis fail from _DL_TMPDIR="/var/tmp/sym link" DATALAD_TESTS_OBSCURE_PREFIX=- DATALAD_LOG_LEVEL=INFO DATALAD_LOG_TRACEBACK=1 DATALAD_LOG_VMEM=1 </summary> 

```shell
======================================================================
ERROR: datalad.plugin.tests.test_plugins.test_wtf
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/tests/utils.py", line 554, in _wrap_with_tree
    return t(*(arg + (d,)), **kw)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/plugin/tests/test_plugins.py", line 75, in test_wtf
    wtf(dataset=path, on_failure="ignore")
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/interface/utils.py", line 494, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/interface/utils.py", line 482, in return_func
    results = list(results)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/interface/utils.py", line 413, in generator_func
    allkwargs):
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/datalad/interface/utils.py", line 586, in _process_results
    res_lgr(msg, *msgargs)
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 1297, in warning
    self._log(WARNING, msg, args, **kwargs)
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 1421, in _log
    self.handle(record)
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 1431, in handle
    self.callHandlers(record)
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 1493, in callHandlers
    hdlr.handle(record)
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 861, in handle
    self.emit(record)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/nose/plugins/logcapture.py", line 82, in emit
    self.buffer.append(self.format(record))
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 836, in format
    return fmt.format(record)
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 573, in format
    record.message = record.getMessage()
  File "/opt/python/3.5.6/lib/python3.5/logging/__init__.py", line 336, in getMessage
    msg = msg % self.args
TypeError: not enough arguments for format string
-------------------- >> begin captured logging << --------------------
datalad.network: WARNING: RSS/VMS: 113148/392636 kB ...>network:634  ParseResults contains params '&%b5 ΔЙקم๗あ.datc ', which will be ignored
```
</details>

and the same in the abominable `DATALAD_LOG_LEVEL=2 DATALAD_LOG_TARGET=/dev/null DATALAD_TESTS_PROTOCOLREMOTE=1 DATALAD_TESTS_DATALADREMOTE=1 DATALAD_LOG_CWD=1 DATALAD_LOG_OUTPUTS=1 DATALAD_LOG_ENV=1 DATALAD_LOG_STDIN=1 DATALAD_TESTS_UI_BACKEND=console DATALAD_TESTS_OBSCURE_PREFIX=- DATALAD_SEED=1 GIT_AUTHOR_DATE="Thu, 07 Apr 2005 22:13:13 +0200" GIT_AUTHOR_NAME=blah GIT_AUTHOR_EMAIL=committer@example.com GIT_COMMITTER_DATE="Thu, 07 Apr 2005 22:13:13 +0200" GIT_COMMITTER_NAME=blah GIT_COMMITTER_EMAIL=committer@example.com`

So -- having `%` as embedded with filename in the logstring somewhere manages to get into the logs and thus cause such troubles!  yet to look into it

**Comment by yarikoptic** [comment]
grr -- failed to reproduce locally with
```TMPDIR=/tmp/sym\ link DATALAD_TESTS_OBSCURE_PREFIX=- DATALAD_LOG_LEVEL=INFO DATALAD_LOG_TRACEBACK=1 DATALAD_LOG_VMEM=1 python -m nose -s -v datalad/plugin/tests/test_plugins.py
```
and thanks to all the decorators and "message passing" I have no clue from above stack about the origin of the problem besides that it is due to some warning (but not that one obviously which posted fine) somewhere while rendering results records.... I think I will just safeguard that invocation of the logger since it better not puke like that

**Comment by yarikoptic** [comment]
ok, still don't know why it failed to replicate locally but I think I know what could have lead to it, so pushed the fix (we need to sanitize path before including into msg).

**Comment by yarikoptic** [comment]
I want to push more tune ups
- RF get_obscure to simplify logic -- just add trailing spaces into parts
- RF get_obscure a bit more to remove windows specific code by testing obscure also on `mkdir`
- add `{}` to most obscure components.  So sample local obscure name without prefix is `' |;&%b5{}ΔЙקم๗あ .datc '`... 
- now I realized that the most evil would be do add `'` and `"` into parts! that makes it into `' |;&%b5{}\'"ΔЙקم๗あ .datc '` on my system... will run tests locally now to see what booms

I could push here, but this one is primarily a BF and IMHO it is already doing BFing it needed to do, or could initiate a new PR on top (after or before we merge this one).  @kyleam - what do you think?

**Comment by kyleam** [comment]
> I want to push more tune ups [...] @kyleam - what do you think?

I haven't looked at the fixups since I last reviewed, but as I reviewer I would very much prefer that PRs aren't a frequently moving target.

**Comment by yarikoptic** [comment]
ok, I will leave this one alone, and will submit a follow up whenever this one is merged (to not build on top of it with a more complicated diff etc) -- I do not think there is any urgency to that matter.

**Pull request merged**
